boards/BMS/Src/App/states/App_InitState.c

```diff
@@ -30,24 +32,32 @@ static void InitStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
     if (App_AllStatesRunOnTick100Hz(state_machine))
     {
-        struct BmsWorld *world = App_SharedStateMachine_GetWorld(state_machine);
+        struct BmsWorld *      world   = App_SharedStateMachine_GetWorld(state_machine);
+        struct TractiveSystem *ts      = App_BmsWorld_GetTractiveSystem(world);
+        struct Airs *          airs    = App_BmsWorld_GetAirs(world);
+        struct Charger *       charger = App_BmsWorld_GetCharger(world);
+
+        bool is_charger_connected = App_Charger_IsConnected(charger);
+
 #ifndef BSPD_DEMO_MODE
         struct TractiveSystem *ts   = App_BmsWorld_GetTractiveSystem(world);
         struct Airs *          airs = App_BmsWorld_GetAirs(world);
         // don't allow pre_charge if in BSPD_DEMO_MODE
         if (App_Airs_IsAirNegativeClosed(airs) && (App_TractiveSystem_GetVoltage(ts) < TS_DISCHARGED_THRESHOLD_V))
         {
-            App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
+            // If the charger is connected, the CAN message must also be set
+            // to continue into Pre-Charge State, then Charge State
+            if (is_charger_connected && App_CanRx_DEBUG_ChargingSwitch_StartCharging_Get())
+            {
+                App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
+            }
+            // If charger isn't connected, go into Pre-Charge State, then Drive State
+            else if (!is_charger_connected)
+            {
+                App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
+            }
         }
 #endif
-        //      #else?
-        struct HeartbeatMonitor *hb_monitor    = App_BmsWorld_GetHeartbeatMonitor(world);
-        const bool               is_missing_hb = !App_SharedHeartbeatMonitor_Tick(hb_monitor);
-        App_CanTx_BMS_Warnings_MissingHeartBeat_Set(is_missing_hb);
-        if (is_missing_hb)
-        {
-            App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
-        }
```

Comment:
You should revert this change here, I don't think this is supposed to be part of your PR